src/compiler/types.ts

```diff
@@ -1064,6 +1064,7 @@ module ts {
         getGlobalDiagnostics(): Diagnostic[];
         getSemanticDiagnostics(sourceFile?: SourceFile): Diagnostic[];
         getDeclarationDiagnostics(sourceFile?: SourceFile): Diagnostic[];
+        /* @internal */ getCompilerOptionsDiagnostics(): Diagnostic[];

```

Comment:
I'm not just concerned about external clients.  I'm also concerned about our own code :)

This is now more confusing for us in terms of knowing which of these functions we should be calling.  

Also, do we need to fix up anything with 'emit' in the LS case?  If the user has hte option set to just transpile, should we be changing how we get errors?